cpp/CMakeLists.txt

```
@@ -83,31 +83,44 @@ include(cmake/thirdparty/get_nvtx.cmake)
 # ##################################################################################################
 # * library targets --------------------------------------------------------------------------------
 
-add_library(rmm INTERFACE)
+add_library(rmm SHARED src/aligned.cpp src/cuda_device.cpp src/cuda_stream_pool.cpp
```

We want to be able to build `RMM` as static for SPARK RAPIDS builds since we want cudf + dependencies to be static
```suggestion
add_library(rmm src/aligned.cpp src/cuda_device.cpp src/cuda_stream_pool.cpp
```